lib/parachain/collation_fetching.go

```diff
@@ -0,0 +1,86 @@
+package parachain
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// CollationFetchingRequest represents a request to retrieve
+// the advertised collation at the specified relay chain block.
+type CollationFetchingRequest struct {
+	// Relay parent we want a collation for
+	RelayParent common.Hash
+
+	// Parachain id of the collation
+	ParaID ParaID
+}
+
+// Encode returns the SCALE encoding of the CollationFetchingRequest
+func (c *CollationFetchingRequest) Encode() ([]byte, error) {
+	return scale.Marshal(*c)
+}
+
+// CollationFetchingResponse represents a response sent by collator
+type CollationFetchingResponse scale.VaryingDataType
+
+// Collation represents a requested collation to be delivered
+type Collation struct {
+	CandidateReceipt CandidateReceipt `scale:"1"`
+	PoV              PoV              `scale:"2"`
+}
+
+// PoV represents a Proof-of-Validity
+type PoV struct {
```

Comment:
This isnt a vdt is it? Why struct with just one element?

Reply:
I think I can change it to `type PoV []byte`